src/components/ImageButton.tsx

```diff
@@ -0,0 +1,41 @@
+import { useNavigation } from '@react-navigation/native';
+import React from 'react';
+import { StyleSheet } from 'react-native';
+
+import { Icon, colors, texts } from '../config';
+
+import { Button } from './Button';
+import { Wrapper } from './Wrapper';
+
+type TImageButton = {
+  icon?: React.ReactElement;
+  params?: any;
+  routeName: string;
+  title?: string;
+};
+
+export const ImageButton = ({ button }: { button: TImageButton }) => {
+  const { icon, params, routeName, title = texts.sue.homeCarousel.button } = button;
+  const navigation = useNavigation();
+
+  if (!params || !routeName) {
+    return null;
+  }
+
+  return (
+    <Wrapper style={styles.noPadding}>
+      <Button
+        icon={icon || <Icon.Plus color={colors.lightestText} />}
```

Comment:
what if there shouldn't be an icon at all? i suggest not having a fallback. if one wants an icon, it should be configured.

Reply:
Hmm. I'll change that. At the moment we have an icon in all cases. I'll make it updatable from the server whether we want an icon or not. 👍

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/pull/957/commits/8a32da27e4cd615ba5eae900256f46644863e75d